projects/js-packages/connection/components/in-place-connection/index.jsx

```diff
@@ -0,0 +1,128 @@
+/**
+ * External dependencies
+ */
+import React, { useEffect, useRef } from 'react';
+import { __ } from '@wordpress/i18n';
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+
+/**
+ * The in-place connection component.
+ *
+ * @param {object} props -- The properties.
+ * @param {string} props.title -- Element title.
+ * @param {boolean} props.isLoading -- Whether the element is still loading.
+ * @param {string|number} props.width -- Iframe width.
+ * @param {string|number} props.height -- Iframe height.
+ * @param {boolean} props.hasConnectedOwner -- Whether the site has connection owner connected.
+ * @param {boolean} props.scrollToIframe -- Whether we need to auto-scroll the window upon element rendering.
+ * @param {string} props.connectUrl -- The connection URL.
+ * @param {Function} props.onComplete -- The callback to be called upon complete of the connection process.
+ * @param {string} props.source -- Component location identifier passed to WP.com.
```

Comment:
At first glance I found this a bit confusing as I was expecting an override or something related to the iframe's source.
Do you think it would make sense to rename it to `triggeredBy` or something similar?

Reply:
You're right, `source` is kind of ambiguous.
`triggeredBy` is better, but it assumes that the calling component is important, while we need the location in the UI, and don't care that much what exactly triggered it. We use the value to adjust the iframe styling, and it only cares about where the iframe will be displayed.
I don't have any better ideas though, so I'll need think about it.
If you have any thoughts, please share 🙂

Reply:
`location` maybe? We can come back to this later as it involves changes on WPCOM as well.

Reply:
based on the description "location identifier"
I would tend to agree with location

Reply:
Renamed to `location` 🙂